Formula/biber@2.16.rb

```diff
@@ -0,0 +1,584 @@
+  end
+  resource "XML::LibXML" do
+    url "https://cpan.metacpan.org/authors/id/S/SH/SHLOMIF/XML-LibXML-2.0207.tar.gz"
+    sha256 "903436c9859875bef5593243aae85ced329ad0fb4b57bbf45975e32547c50c15"
+  end
+  resource "XML::LibXML::Simple" do
+    url "https://cpan.metacpan.org/authors/id/M/MA/MARKOV/XML-LibXML-Simple-1.01.tar.gz"
+    sha256 "cd98c8104b70d7672bfa26b4513b78adf2b4b9220e586aa8beb1a508500365a6"
+  end
+  resource "XML::LibXSLT" do
+    url "https://cpan.metacpan.org/authors/id/S/SH/SHLOMIF/XML-LibXSLT-1.99.tar.gz"
+    sha256 "127e17a877fb61e47b9e8b87bf8daad31339a62a00121f9751d522b438b0f7f0"
+  end
+  resource "XML::Writer" do
+    url "https://cpan.metacpan.org/authors/id/J/JO/JOSEPHW/XML-Writer-0.900.tar.gz"
+    sha256 "73c8f5bd3ecf2b350f4adae6d6676d52e08ecc2d7df4a9f089fa68360d400d1f"
+  end
+  resource "autovivification" do
+    url "https://cpan.metacpan.org/authors/id/V/VP/VPIT/autovivification-0.18.tar.gz"
+    sha256 "2d99975685242980d0a9904f639144c059d6ece15899efde4acb742d3253f105"
+  end
+
+  def install
+    ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
+    ENV.prepend_path "PERL5LIB", libexec/"lib"
+    ENV["PERL_MM_USE_DEFAULT"] = "1"
+    ENV["OPENSSL_PREFIX"] = Formula["openssl@1.1"].opt_prefix
+
+    resources.each do |r|
+      r.stage do
+        # fix libbtparse.so linkage failure on Linux
+        if r.name == "Text::BibTeX" && OS.linux?
+          inreplace "inc/MyBuilder.pm",
+                    "-lbtparse",
+                    "-Wl,-rpath,#{libexec}/lib -lbtparse"
+        end
+
+        if File.exist? "Makefile.PL"
+          system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+          system "make"
+          system "make", "install"
+        else
+          system "perl", "Build.PL", "--install_base", libexec
+          system "./Build"
+          system "./Build", "install"
+        end
+      end
+    end
+
+    system "perl", "Build.PL", "--install_base", libexec
+    system "./Build"
+    system "./Build", "install"
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", PERL5LIB: ENV["PERL5LIB"])
+    man1.install libexec/"man/man1/biber.1"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/biber --version")
+
+    (testpath/".latexmkrc").write <<~EOS
+      $biber='#{bin}/biber %O %S'
+    EOS
+
+    (testpath/"test.bib").write <<~EOS
+      @online{test1,
+        author    = "Donald Knuth",
+        title     = "Knuth: Computers and Typesetting",
+        url       = "http://www-cs-faculty.stanford.edu/~uno/abcde.html",
+        keywords  = "latex,knuth"
+      }
+    EOS
+
+    (testpath/"test.tex").write <<~EOS
+      \\documentclass{article}
+      \\usepackage{biblatex}
+      \\addbibresource{test.bib}
+      \\begin{document}
+        Test \\cite{test1}
+        \\printbibliography
+      \\end{document}
+    EOS
+
+    system "latexmk", "-pdf", "test.tex"
```

Comment:
Pulling in texlive for the test would be quite a large dependency given it's current size, wouldn't it?

Reply:
I think the problem we have with the current formula is that the functionality of `biber` depends on having a LaTeX installation available, and we generally try not to ship formulae that are lacking dependencies they need to function.  The tricky thing with `biber` is that there are other ways you can install LaTeX than our `texlive` formula, some of which do have smaller installation footprints.  But these alternative installation methods don't work for all of our users (especially if they don't have root access, particularly on Linux), so we probably want to provide at least one version of `biber` that we know will work with `texlive`, even if it means having to depend on `texlive` to test it.